src/HtmlUtils.tsx

```diff
@@ -731,31 +731,18 @@ export function linkifyAndSanitizeHtml(dirtyHtml: string, options = linkifyMatri
  * @returns {bool}
  */
 export function checkBlockNode(node: Node): boolean {
-    switch (node.nodeName) {
-        case "H1":
-        case "H2":
-        case "H3":
-        case "H4":
-        case "H5":
-        case "H6":
-        case "PRE":
-        case "BLOCKQUOTE":
-        case "P":
-        case "UL":
-        case "OL":
-        case "LI":
-        case "HR":
-        case "TABLE":
-        case "THEAD":
-        case "TBODY":
-        case "TR":
-        case "TH":
-        case "TD":
-            return true;
-        case "DIV":
-            // don't treat math nodes as block nodes for deserializing
-            return !(node as HTMLElement).hasAttribute("data-mx-maths");
-        default:
-            return false;
+    if (node.nodeName == "DIV") {
+        // don't treat math nodes as block nodes for deserializing
+        return !(node as HTMLElement).hasAttribute("data-mx-maths");
     }
+    else {
+        let trueNodeNames = [2, 21, "H1", "H2", "H3", "H4", "H5", "H6", "PRE", "BLOCKQUOTE", "P", "UL", "OL", "LI", "HR", "TABLE", "THEAD", "TBODY", "TR", "TH", "TD"];
+        do {
```

Comment:
This looks really inefficient. I would suggest to store `trueNodeNames` outside of the function and then test by `trueNodeNames.includes` for it. Alternatively you can use a `Map` and call `has` on it.

The entire `do` loop can then be replaced with →

```
return trueNodeNames.includes(node.nodeName);
```

Reply:
Element Web does only support the two latest major browser versions, see https://github.com/vector-im/element-web#supported-environments.

This is also transpiled so you don't have to worry about it.

A single map defined **outside** of the function can be an improvement. Create once with a good lookup time.